pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -824,10 +827,13 @@ func (k *KSMCheck) sendTelemetry(s sender.Sender) {
 	// reset the cache for the next check run
 	defer k.telemetry.reset()
 
-	s.Gauge(ksmMetricPrefix+"telemetry.metrics.count.total", float64(k.telemetry.getTotal()), "", k.instance.Tags)
-	s.Gauge(ksmMetricPrefix+"telemetry.unknown_metrics.count", float64(k.telemetry.getUnknown()), "", k.instance.Tags) // useful to track metrics that aren't mapped to DD metrics
+	s.Gauge(ksmMetricPrefix+"telemetry.metrics.count.total", float64(k.telemetry.getTotal()), "", util.CopyArray(k.globalTags))
+	s.Gauge(ksmMetricPrefix+"telemetry.unknown_metrics.count", float64(k.telemetry.getUnknown()), "", util.CopyArray(k.globalTags))
 	for resource, count := range k.telemetry.getResourcesCount() {
-		s.Gauge(ksmMetricPrefix+"telemetry.metrics.count", float64(count), "", append(k.instance.Tags, "resource_name:"+resource))
+		tags := make([]string, len(k.globalTags)+1)
+		copy(tags, k.globalTags)
+		tags[len(k.globalTags)] = "resource_name:" + resource
+		s.Gauge(ksmMetricPrefix+"telemetry.metrics.count", float64(count), "", tags)
```

Comment:
Can we use the same pattern and use `util.CopyArray` for all the `s.Gauge(…)` invocations ?

Reply:
Indeed. I didn't do it because it would perform an extra allocation but we can favor readability here